lib/plsql/variable.rb

```diff
@@ -47,7 +47,7 @@ def value=(new_value)
 
     def metadata(type_string)
       case type_string
-      when /^(VARCHAR|VARCHAR2|CHAR|NVARCHAR2|NCHAR)(\((\d+)[\s\w]*\))?$/
+      when /^(VARCHAR|VARCHAR2|CHAR|NVARCHAR2|NCHAR)(?:\s+)?(\((?:\s+)?(\d+)[\s\w]*\))?$/
```

Comment:
I'm wondering if this can be simplified a bit. E.g. what about using `\s*` instead of `(\s+)?`?